src/libdistgen.cpp

```diff
@@ -175,7 +191,34 @@ double distgend_is_membound_scaled(distgend_configT config) {
 static void *thread_benchmark(void *arg) {
 	thread_argsT *thread_args = (thread_argsT *)arg;
 	double *ret = (double *)calloc(1, sizeof(double));
+#ifdef __QNXNTO__
```

Comment:
Why don't you move this into a separate function `set_affinity` and call it for both the linux and the QNX variant at this place? We shouldn't run into NUMA effects calling it here, should we?